app/controllers/quizzes/quiz_submissions_controller.rb

```diff
@@ -37,6 +37,8 @@ def index
 
   # submits the quiz as final
   def create
+    url_params = previewing_params
+    url_params[:module_item_id] = params[:module_item_id]
```

Comment:
I don't know enough about the code to say whether this is the right way to fix it, but I will say that it doesn't "feel" like this fix is the way that Canvas generally handles this sort of thing.  I'm fine with it for now until we really start to understand their coding conventions and general design patterns.